libdino/src/service/file_manager.vala

```diff
@@ -46,6 +46,15 @@ public class FileManager : StreamInteractionModule, Object {
         return ret;
     }
 
+    private string sanitize_filename(string filename) {
+#if _WIN32
+        GLib.Regex regex = new GLib.Regex("[<>:\"/\\|?*]");
+#else
+        GLib.Regex regex = new GLib.Regex("[/\\x{00}]");
```

Comment:
@Alphastaire In Vala, strings are null-terminated. This check for `'\0'` is useless. Leave only the check for `/`. And also squash all the commits into one. Thanks.